### PR TITLE
[Linux/x86] Enable CI Linux/x86 build

### PR DESCRIPTION
**Pull request by hseok-oh**
Make new job: Linux/x86 build (debug/release)

- Triggered daily
- Docker for crossbuild
- Use pre-generated rootfs

related issue: dotnet/coreclr#9903

### CODE REVIEW

**Comment by hseok-oh** [comment]
@dotnet-bot test ci please

**Comment by hseok-oh** [comment]
@dotnet-bot test ci please

**Comment by hseok-oh** [comment]
@dotnet-bot test ci please

**Comment by hseok-oh** [comment]
@dotnet-bot test ci please

**Comment by hseok-oh** [comment]
@dotnet-bot help

**Comment by hseok-oh** [comment]
@dotnet-bot test innerloop Ubuntu16.04 arm Debug
@dotnet-bot test innerloop Ubuntu16.04 arm Release

**Comment by hseok-oh** [comment]
Successfully generated new job for Linux/x86.
I've check build script indirectly using Linux/arm32 build: used temporary modified arm32_ci_script.sh to build x86 (same code as x86_ci_script.sh).

https://ci.dot.net/job/dotnet_corefx/job/master/job/ubuntu16.04_arm_cross_release_prtest/12/console
https://ci.dot.net/job/dotnet_corefx/job/master/job/ubuntu16.04_arm_cross_debug_prtest/11/console

Finished as failure, but it was because of tar operation of result (tried to find arm build result, but couldn't  find), not build operation.

**Comment by hseok-oh** [comment]
It seems ready to merge. PTAL @mellinoe @gkhanna79

File: cross/x86_ci_script.sh [hunk 1]
```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+
+function exit_with_error {
+    set +x
+
+    local errorMessage="$1"
+    local printUsage=$2
```
Comment (mellinoe) [inline cross/x86_ci_script.sh]: `printUsage` doesn't seem to be used anywhere.

File: netci.groovy [hunk 2]
```diff
@@ -513,6 +513,41 @@ def buildArchConfiguration = ['Debug': 'x86',
     } // targetGroup
 } // isPR
 
+// **************************
+// Define Linux x86 builds. These jobs run daily and results will be used for CoreCLR test
+// TODO: innerloop & outerloop testinig & merge to general job generation routine
```
Comment (mellinoe) [inline netci.groovy]: Typo: ~~testinig~~ testing

File: cross/x86_ci_script.sh [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        exit_with_error "$errorMessage" $printUsage
+    fi
+}
+
+# Cross builds corefx using Docker image
+function cross_build_native_with_docker {
+    __currentWorkingDirectory=`pwd`
+
+    # choose Docker image
+    __dockerImage=" hseok82/dotnet-buildtools-prereqs:ubuntu1604_cross_prereqs_v3_x86"
```
Comment (mellinoe) [inline cross/x86_ci_script.sh]: I don't have any specific concerns, but I'm curious what our policy around docker images and registries is. Should we try to put these into a central trusted location or are we fine with using user repositories for CI jobs like this?

@mmitche @chcosta @karelz  (Not sure who to tag about this)

Comment (karelz) [inline cross/x86_ci_script.sh]: cc @gkhanna79 @danmosemsft @weshaggard 
Naively, central trusted location sounds like safer approach, but I don't have enough info about Core the infra.

Comment (gkhanna79) [inline cross/x86_ci_script.sh]: The goal is to have a central location and associated automation to build docker images that not only we use consistently across the board but also enable community to contribute to their definitions.

@MichaelSimons has a work item assigned to him in this regard. For now, the current approach is fine. @hseok-oh can you please file an issue against yourself to switch to official image once that is available?

Comment (mellinoe) [inline cross/x86_ci_script.sh]: Sounds good to me. I just wanted to make sure I knew what our current thinking was.

**Comment by mellinoe** [comment]
I left a couple of minor comments, but other than those it LGTM.

File: cross/x86_ci_script.sh (hunk 3 above)
Comment (hseok-oh) [inline cross/x86_ci_script.sh]: Sure. I'll file a new issue. 
Its temporary docker image will be changed to official one when that is ready to use.

**Comment by hseok-oh** [comment]
@mellinoe I've fixed based on your comments. Thanks.

File: cross/x86_ci_script.sh [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+    echo "ERROR: $errorMessage"
+    exit 1
+}
+
+#Exit if input string is empty
+function exit_if_empty {
+    local inputString="$1"
+    local errorMessage="$2"
+    local printUsage=$3
```
Comment (mellinoe) [inline cross/x86_ci_script.sh]: This variable still isn't being used, as far as I can tell.

File: cross/x86_ci_script.sh [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+    if [ -z "$inputString" ]; then
+        exit_with_error "$errorMessage" $printUsage
```
Comment (mellinoe) [inline cross/x86_ci_script.sh]: exit_with_error isn't reading the second parameter passed to it anyways. Just remove the `printUsage` stuff?

**Comment by hseok-oh** [comment]
@mellinoe Fixed. PTAL.

**Comment by stephentoub** [comment]
@mellinoe, can/should this be merged? Thanks.

**Comment by mellinoe** [comment]
@hseok-oh Sorry for the delay, merging.

**Pull request merged**